### PR TITLE
feat: Improve code quality 

### PR DESCRIPTION
**Pull request by bokket**
Signed-off-by: bokket <3100563328@qq.com>
This pr fixed :#85
My machine verison and make build success
image
image
But now the test is not cross-platform (bitter laugh), gtest I did not whole, may find other installation, such as https://github.com/onqtam/doctest

### CODE REVIEW

**Comment by xcaptain** [comment]
I think to make semantic pull request check pass you need to modify all your commit messages and prefix a fix or feat, because the current setting is `titleAndCommits: true`

**Comment by hsluoyz** [comment]
@bokket squash into only one commit

**Comment by bokket** [comment]
> @bokket压入一次提交

Ths，it's done.

File: casbin/exception/unsupported_operation_exception.h [hunk 1]
```diff
@@ -9,7 +9,8 @@ using namespace std;
 class UnsupportedOperationException{
     string error_message;
     public:
-        UnsupportedOperationException(string error_message);
+        explicit UnsupportedOperationException(string& error_message);
+        explicit UnsupportedOperationException(const char* error_message);
```
Comment (xcaptain) [inline casbin/exception/unsupported_operation_exception.h:13]: same as above, I think we should forbidden using `char *` in this project.

File: casbin/ip_parser/parser/byte.h [hunk 2]
```diff
@@ -1,7 +1,8 @@
 #ifndef IP_PARSER_PARSER_BYTE
 #define IP_PARSER_PARSER_BYTE
 
-typedef unsigned short int byte;
+using namespace std;
+typedef unsigned short int Byte;
```
Comment (xcaptain) [inline casbin/ip_parser/parser/byte.h:5]: short int is 16 bits, why name it `Byte`, didn't notice this before

Comment (bokket) [reply casbin/ip_parser/parser/byte.h:5]: It is described by the idea of @​​divy9881 and conflicts with c++17 std::byte

File: casbin/config/config.cpp [hunk 3]
```diff
@@ -77,14 +77,15 @@ void Config :: ParseBuffer(istream* buf){
...
         else {
-            vector<string> option_val = Split(line, string("="), 2);
+            vector<string> option_val = Split(line, "=",2);
             if (option_val.size() != 2) {
-                char* error = new char;
-                sprintf(error, "parse the content error : line %d , %s = ? ", line_num, option_val[0].c_str());
-                throw IllegalArgumentException(string(error));
+                //char* error = new char;
+                string error;
+                sprintf(const_cast<char*>(error.data()), "parse the content error : line %d , %s = ? ", line_num, option_val[0].c_str());
```
[the remark this answers is not in the document]
Comment (bokket) [reply casbin/config/config.cpp:87]: I don't know if msvc supports this feature perfectly, but stringstream can be used

File: casbin/error/error.h [hunk 4]
```diff
@@ -10,11 +10,18 @@ class Error{
         static Error NIL;
         string err;
 
-        Error(string error_message){
-            err = error_message;
+        explicit Error(string& error_message):err(error_message) {
         }
 
-        string toString(){
+        explicit Error(const char* error_message):err(error_message) {
```
[the remark this answers is not in the document]
Comment (bokket) [reply casbin/error/error.h:16]: ok

**Comment by hsluoyz** [comment]
@divy9881 @xcaptain please review.

**Comment by hsluoyz** [comment]
@bokket why close?

**Comment by bokket** [comment]
> @bokket为什么要关门？

I don't know which suggestion you agree with more, if based on #88, this is a new start

**Pull request closed without merging**